assets/install-rhcos.sh

```
@@ -1,53 +1,142 @@
+    local boot_fifo_path="/sys/bus/platform/devices/MLNXBF04:00/bootfifo"
+    local cfg_file
+    if [ -e "${boot_fifo_path}" ]; then
+        cfg_file=$(mktemp)
+        ilog "Trying to copy ignition..."
+        dd if=${boot_fifo_path} of=${cfg_file} bs=4096 count=1000 > /dev/null 2>&1
+
+        # Check the .xz signature {0xFD, '7', 'z', 'X', 'Z', 0x00} and extract the config file from it.
+        local offset
+        offset=$(strings -a -t d ${cfg_file} | grep -m 1 "7zXZ" | awk '{print $1}')
+        
+        if [ -s "${cfg_file}" -a ."${offset}" != ."1" ]; then
```

Shellcheck (0.10.0)

[warning] Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.

(SC2166)